src/Symfony/Bundle/WebProfilerBundle/Controller/ProfilerController.php

```diff
@@ -395,6 +395,34 @@ public function phpinfoAction()
     }
 
     /**
+     * Displays the source of a file.
+     *
+     * @return Response A Response instance
+     *
+     * @throws NotFoundHttpException
+     */
+    public function openAction(Request $request)
+    {
+        if (null === $this->profiler) {
+            throw new NotFoundHttpException('The profiler must be enabled.');
+        }
+
+        $this->profiler->disable();
+
+        $file = $request->query->get('file');
+        $line = $request->query->get('line');
+
+        if (0 === strpos($file, '/..') || !is_readable($filename)) {
```

Comment:
`/..` is not enough (e.g. on Windows, or single dot). What about `preg_patch("'^([/\\]|\.\.?([/\\]|$))'", $file)`?